Projects/Steam/Steam/Steam.cpp

```diff
@@ -58,7 +73,7 @@ void FSteam::MainMenu()
 	int Option;
 	std::cin >> Option;
 
-	Option = ValidateInt(Option, 1, 4);
+	ValidateInt(Option, 1, 4);
```

Comment:
This method has been working fine, but I have two comments:
1. Why is it necessary to have a validator rather than a getter? Instead of `ValidateInt()` you could have a `GetValidIntInput()` or something like that. That way, the first `cin >> Option` can occur inside the method itself and so you don't have to do it every time you want to get an int input.

2. Ouput parameters are useful in some scenarios, like for example when you want to return a value but also return whether or not the returned value is valid, so you use a bool return type and an out parameter for the other type you want to return.

However, they are not the standard way of returning stuff, that's what the return type is for. With this in mind, although this method works fine, I believe that it would be clearer if you used the return type, the way you were doing before.

Ouput parameters should be special cases, rather than the rule. Especially with built-in types.